.github/workflows/ci.yaml

```diff
@@ -0,0 +1,125 @@
+name: CI
+
+on: [push]
+
+jobs:
+  cvc4:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Inspect
+      run: |
+         echo $PWD
+         find .
+    - name: Get CVC4
+      run: |
+        git clone -b eqrange-quant --single-branch https://github.com/GaloisInc/vadd-CVC4.git .
+    - name: Get Antlr
+      run: ./contrib/get-antlr-3.4
+    - name: Configure
+      run: ./configure.sh production --static --proofs --prefix=$PWD/dist
+    - name: Build
+      working-directory: build
+      run: make -j 8
+    - name: Package
+      run: |
+        find build -name cvc4
+        readelf -d build/bin/cvc4
+    - name: cvc4 artifact
+      uses: actions/upload-artifact@v1
+      with:
+        name: cvc4
+        path: build/bin/cvc4
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        # NOTE: must list all targets in `include` here to activate them
+        name: ["Linux 8.8.3"]
+        include:
+          - name: Linux 8.8.3
+            ghc-ver: 8.8.3
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        submodules: true
+    - name: Get GHC
+      run: |
+        sudo add-apt-repository ppa:hvr/ghc
+        sudo apt-get update
+        sudo apt-get install cabal-install-3.0 ghc-${{ matrix.ghc-ver }} llvm
+    - name: Cache
+      uses: actions/cache@v1
+      with:
+        path: home/.cabal/store/ghc-${{ matrix.ghc-ver }}
+        key: ${{ matrix.name }}-build-${{ github.sha }}
```

Comment:
This key will only ever match if you retry a failed build. I'd suggest just using something like `${{ runner.os}}-${{ matrix.ghc }}-cabal-store`.

If you want to get fancy with it, you could have the primary key be the hash of the `cabal.project.freeze` file and then the fallback would be the "generic" cabal-store key.

eg:

```yaml
key: ${{ matrix.name }}-build-${{ hashFiles('**/cabal.project.freeze') }}
restore-keys: ${{ runner.os}}-${{ matrix.ghc }}-cabal-store
```

Reply:
runner.os should always be the same right?

Reply:
In your case, yes. If you decide to test on windows or another OS for some reason, then it won't be. Might be unnecessary future-proofing, but it depends. It's very necessary for Cryptol, but not every project shares Cryptol's concerns.